padl/transforms.py

```diff
@@ -882,11 +936,19 @@ def infer_apply(self, inputs):
         if use_preprocess:
             inputs = preprocess.pd_call_in_mode(inputs, mode='infer', ignore_grad=True)
         if pd_device != 'cpu':
-            inputs = _move_to_device(inputs, pd_device)
+            inputs = _move_to_device(inputs, self.pd_device)
```

Comment:
Change `self.pd_device` to local variable `pd_device` since it's already been defined.